### PR TITLE
[docs] [WIP] Add a blog post

### PR DESCRIPTION
**Pull request by AlexanderMoskovkin**

### CODE REVIEW

File: docs/articles/blog/2017-6-00-testcafe-v0-16-0-released.md [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+
+### ⚙ TypeScript support ([#408](https://github.com/DevExpress/testcafe/issues/408))
+
+### ⚙ Support running in Chrome in headless mode and in device emulation ([#1417](https://github.com/DevExpress/testcafe/issues/1417))
+
+### ⚙ Support HTML5 Drag and Drop ([#897](https://github.com/DevExpress/testcafe/issues/897))
+
+### ⚙ Fixed URL for opening remote browsers ([#1476](https://github.com/DevExpress/testcafe/issues/1476))
+
+### ⚙ No TestCafe UI on screenshots created during testing ([#1357](https://github.com/DevExpress/testcafe/issues/1357))
```
Comment (miherlosev) [inline docs/articles/blog/2017-6-00-testcafe-v0-16-0-released.md]: It is more bug than enhancements. So, I propose to move this to 'Bug Fixes' section

File: docs/articles/blog/2017-6-7-testcafe-v0-16-0-released.md [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+
+### ⚙ TypeScript support ([#408](https://github.com/DevExpress/testcafe/issues/408))
+
+In this release, we have added an ability to write tests in [TypeScript](https://www.typescriptlang.org/). Using of TypeScript brings you all the advantages of strongly typed languages: rich coding assistance, painless scalability, check-as-you-type code verification and much more.
+
+TestCafe bundles TypeScript declaration file with the npm package, so you have no need to install any additional packages.
+
+Just create a `.ts` file with the
+
+<!-- Use `js` instaed of `ts` for this code block for proper code highlighting -->
```
Comment (helen-dikareva) [inline docs/articles/blog/2017-6-7-testcafe-v0-16-0-released.md]: type `instaed`

File: docs/articles/blog/2017-6-7-testcafe-v0-16-0-released.md [hunk 3]
```diff
@@ -0,0 +1,74 @@
...
+import { Selector } from 'testcafe';
+```
+
+and write your test.
+
+For details, see [TypeScript Support](https://devexpress.github.io/testcafe/documentation/test-api/typescript-support.html)
+
+### ⚙ Support running in Chrome in headless mode and in device emulator ([#1417](https://github.com/DevExpress/testcafe/issues/1417))
+
+Now TestCafe allows you to run your tests in Google Chrome in the headless mode and the device emulation mode.
```
Comment (lexkazakov) [inline docs/articles/blog/2017-6-7-testcafe-v0-16-0-released.md]: Now TestCafe allows you to run your tests in Google Chrome in headless and device emulation modes.

File: docs/articles/blog/2017-6-7-testcafe-v0-16-0-released.md [hunk 4]
```diff
@@ -0,0 +1,74 @@
...
+
+[Headless mode](https://developers.google.com/web/updates/2017/04/headless-chrome) allows you to run tests in Chrome without any visible UI shell. To run tests in the headless mode use the `:headless` postfix:
```
Comment (lexkazakov) [inline docs/articles/blog/2017-6-7-testcafe-v0-16-0-released.md]: ... To run tests in headless mode, use the `:headless` postfix:

File: docs/articles/blog/2017-6-7-testcafe-v0-16-0-released.md [hunk 5]
```diff
@@ -0,0 +1,74 @@
...
+
+```sh
+testcafe "chrome:headless" tests/sample-fixture.js
+```
+
+Device emulation mode allows you to check how your tests works on mobile devices via Chrome's built-in [device emulator](https://developers.google.com/web/tools/chrome-devtools/device-mode/). To run tests in the device emulation mode specify `emulation:` and [device parameters](https://devexpress.github.io/testcafe/documentation/common-concepts/browser-support.md#available-chrome-options):
```
Comment (lexkazakov) [inline docs/articles/blog/2017-6-7-testcafe-v0-16-0-released.md]: ... To run tests in device emulation mode, specify `emulation:` and [device parameters](https://devexpress.github.io/testcafe/documentation/common-concepts/browser-support.md#available-chrome-options):

File: docs/articles/blog/2017-6-7-testcafe-v0-16-0-released.md [hunk 6]
```diff
@@ -0,0 +1,74 @@
...
+
+We have simplified the format of links that TestCafe generates when you [run tests on remote browsers](https://devexpress.github.io/testcafe/documentation/using-testcafe/common-concepts/browser-support.html#browsers-on-remote-devices).
+
+Now, you have no need to type a unique link for each test run, all the links became constant. So, it is easier now to run tests on a remote device repeatedly: you can run them by navigating a link from your browser history.
+
+## Bug Fixes
+
+* No TestCafe UI on screenshots created during testing ([#1357](https://github.com/DevExpress/testcafe/issues/1357))
+* 'mouseenter' and 'mouseleave' events are not triggered during cursor moving ([#1426](https://github.com/DevExpress/testcafe/issues/1426))
+* Runner's speed option affects the speed of 'doubleClick' action ([#1486](https://github.com/DevExpress/testcafe/issues/1486))
```
Comment (lexkazakov) [inline docs/articles/blog/2017-6-7-testcafe-v0-16-0-released.md]: The runner's speed option affects the speed of 'doubleClick' action ([#1486](https://github.com/DevExpress/testcafe/issues/1486))

File: docs/articles/blog/2017-6-7-testcafe-v0-16-0-released.md [hunk 7]
```diff
@@ -0,0 +1,74 @@
...
+* Press action shortcuts work wrong if input's value ends with '.' or starts with '-.' ([#1499](https://github.com/DevExpress/testcafe/issues/1499))
+* Test report has too small line length on Travis ([#1469](https://github.com/DevExpress/testcafe/issues/1469))
```
Comment (lexkazakov) [inline docs/articles/blog/2017-6-7-testcafe-v0-16-0-released.md]: A test report has too small line length on Travis ([#1469](https://github.com/DevExpress/testcafe/issues/1469))

File: docs/articles/blog/2017-6-7-testcafe-v0-16-0-released.md [hunk 8]
```diff
@@ -0,0 +1,74 @@
...
+
+Starting with this release, TestCafe supports HTML5 drag and drop, so you can test elements with the `draggable` [attribute](http://w3c.github.io/html/editing.html#the-draggable-attribute).
+
+### ⚙ Fixed URL for opening remote browsers ([#1476](https://github.com/DevExpress/testcafe/issues/1476))
+
+We have simplified the format of links that TestCafe generates when you [run tests on remote browsers](https://devexpress.github.io/testcafe/documentation/using-testcafe/common-concepts/browser-support.html#browsers-on-remote-devices).
+
+Now, you have no need to type a unique link for each test run, all the links became constant. So, it is easier now to run tests on a remote device repeatedly: you can run them by navigating a link from your browser history.
+
+## Bug Fixes
```
Comment (miherlosev) [inline docs/articles/blog/2017-6-7-testcafe-v0-16-0-released.md]: add https://github.com/DevExpress/testcafe-hammerhead/issues/1131 to the bug fixes

**Comment by AlexanderMoskovkin** [comment]
FPR

File: docs/articles/blog/2017-6-7-testcafe-v0-16-0-released.md [hunk 9]
```diff
@@ -0,0 +1,75 @@
...
+
+* No TestCafe UI on screenshots created during testing ([#1357](https://github.com/DevExpress/testcafe/issues/1357))
+* 'mouseenter' and 'mouseleave' events are not triggered during cursor moving ([#1426](https://github.com/DevExpress/testcafe/issues/1426))
```
Comment (kirovboris) [inline docs/articles/blog/2017-6-7-testcafe-v0-16-0-released.md]: Please highlight `mouseenter`, `mouseleave` and all others API calls below.

File: docs/articles/blog/2017-6-7-testcafe-v0-16-0-released.md [hunk 10]
```diff
@@ -0,0 +1,75 @@
...
+
+## Bug Fixes
+
+* No TestCafe UI on screenshots created during testing ([#1357](https://github.com/DevExpress/testcafe/issues/1357))
+* `mouseenter` and `mouseleave` events are not triggered during cursor moving ([#1426](https://github.com/DevExpress/testcafe/issues/1426))
+* The runner's speed option affects the speed of `doubleClick` action ([#1486](https://github.com/DevExpress/testcafe/issues/1486))
+* Press action shortcuts work wrong if input's value ends with '.' or starts with '-.' ([#1499](https://github.com/DevExpress/testcafe/issues/1499))
+* A test report has too small line length on Travis ([#1469](https://github.com/DevExpress/testcafe/issues/1469))
+* Service messages with cookies do not have enough time to come to server before a new page is loaded ([testcafe-hammerhead/#1086](https://github.com/DevExpress/testcafe-hammerhead/issues/1086))
+* The window.history.replaceState function is overridden incorrectly ([testcafe-hammerhead/#1146](https://github.com/DevExpress/testcafe-hammerhead/issues/1146))
```
Comment (kirovboris) [inline docs/articles/blog/2017-6-7-testcafe-v0-16-0-released.md]: `window.history.replaceState`

File: docs/articles/blog/2017-6-7-testcafe-v0-16-0-released.md [hunk 11]
```diff
@@ -0,0 +1,75 @@
...
+* Hammerhead crashes if a script file contains a sourcemap comment ([testcafe-hammerhead/#1052](https://github.com/DevExpress/testcafe-hammerhead/issues/1052))
+* The proxy should override the `DOMParser.parseFromString` method ([testcafe-hammerhead/#1133](https://github.com/DevExpress/testcafe-hammerhead/issues/1133))
+* The fetch method should emulate the native behaviour on merging headers ([testcafe-hammerhead/#1116](https://github.com/DevExpress/testcafe-hammerhead/issues/1116))
+* The EventSource requests are broken when used via proxy ([testcafe-hammerhead/#1106](https://github.com/DevExpress/testcafe-hammerhead/issues/1106))
```
Comment (kirovboris) [inline docs/articles/blog/2017-6-7-testcafe-v0-16-0-released.md]: `EventSource `

File: docs/articles/blog/2017-6-7-testcafe-v0-16-0-released.md [hunk 12]
```diff
@@ -0,0 +1,75 @@
...
+* The code processing may cause syntax errors in some cases because of wrong `location` property wrapping ([testcafe-hammerhead/#1101](https://github.com/DevExpress/testcafe-hammerhead/issues/1101))
+* When calling the `fetch` function without parameters, we should return its native result instead of window.Promise.reject ([testcafe-hammerhead/#1099](https://github.com/DevExpress/testcafe-hammerhead/issues/1099))
```
Comment (kirovboris) [inline docs/articles/blog/2017-6-7-testcafe-v0-16-0-released.md]: `window.Promise.reject`

File: docs/articles/blog/2017-6-7-testcafe-v0-16-0-released.md [hunk 13]
```diff
@@ -0,0 +1,75 @@
...
+* The querySelector function is overridden incorrectly ([testcafe-hammerhead/#1131](https://github.com/DevExpress/testcafe-hammerhead/issues/1131))
```
Comment (kirovboris) [inline docs/articles/blog/2017-6-7-testcafe-v0-16-0-released.md]: `querySelector`

File: docs/articles/blog/2017-6-7-testcafe-v0-16-0-released.md [hunk 14]
```diff
@@ -0,0 +1,75 @@
...
+* Press action shortcuts work wrong if input's value ends with '.' or starts with '-.' ([#1499](https://github.com/DevExpress/testcafe/issues/1499))
+* A test report has too small line length on Travis ([#1469](https://github.com/DevExpress/testcafe/issues/1469))
+* Service messages with cookies do not have enough time to come to server before a new page is loaded ([testcafe-hammerhead/#1086](https://github.com/DevExpress/testcafe-hammerhead/issues/1086))
+* The window.history.replaceState function is overridden incorrectly ([testcafe-hammerhead/#1146](https://github.com/DevExpress/testcafe-hammerhead/issues/1146))
+* Hammerhead crashes if a script file contains a sourcemap comment ([testcafe-hammerhead/#1052](https://github.com/DevExpress/testcafe-hammerhead/issues/1052))
+* The proxy should override the `DOMParser.parseFromString` method ([testcafe-hammerhead/#1133](https://github.com/DevExpress/testcafe-hammerhead/issues/1133))
+* The fetch method should emulate the native behaviour on merging headers ([testcafe-hammerhead/#1116](https://github.com/DevExpress/testcafe-hammerhead/issues/1116))
```
Comment (kirovboris) [inline docs/articles/blog/2017-6-7-testcafe-v0-16-0-released.md]: `fetch`

**Pull request merged**